plugins/modules/cloud/misc/terraform.py

```diff
@@ -449,12 +488,70 @@ def main():
     if state == 'present' and module.params.get('parallelism') is not None:
         command.append('-parallelism=%d' % module.params.get('parallelism'))
 
+    def format_args(vars):
+        if isinstance(vars, str):
+            return '"{string}"'.format(string=re.sub('"', '\\"', vars))
```

Comment:
```suggestion
            return '"{string}"'.format(string=vars.replace('"', '\\"'))
```
Though you probably also have to replace `\` by `\\`:
```suggestion
            return '"{string}"'.format(string=vars.replace('\\', '\\\\"').replace('"', '\\"'))
```

Reply:
Good catch, needed to replace `'\\'` with `'\\\\'. Two thoughts:

1. We need to include all the terraform escape sequences as test cases.
2. It would massively help someone using the module from the first time with little guidance how the escape sequences would work....

For example:

Ansible `string_type: 'cli specials"&$%@#*!(){}[]:"" \\'` would translate to Terraform `string_type = "cli specials\"&$%@#*!(){}[]:\"\" \\\\"`